### ISSUE TITLE
Unexpected result of viewing Locations in country zoom

### ISSUE THREAD

**Issue by fdellwing** [matomo-org/matomo#13691]
If you view the locations map on country level you are able to see city, but not state visitors.

![matomo6](https://user-images.githubusercontent.com/22813377/48252074-ffa77700-e403-11e8-811e-f5b1f5ab3446.png)
![matomo7](https://user-images.githubusercontent.com/22813377/48252075-00400d80-e404-11e8-8dc7-67b39d7c0fa2.png)

**Comment by fdellwing** [comment]
@tsteur Can you look into that? The states are clearly defined, as you can see on the right, but are shown as "unknown" on the map.

**Comment by tsteur** [comment]
I'm not so much into the maps. I reckon this might be because of the GeoIP2 update @sgiehl ?

**Comment by sgiehl** [comment]
Yes. That's likely caused by GeoIP2. Our map is quite outdate for some states and (except of USA and Canada) it's defined with FIPS country codes (which was used for GeoIP Legacy). GeoIP2 switched to ISO codes for all countries, which makes it impossible for nearly all countries to match the regions anymore.
We need to recreate all our maps to make the compatible again. But as our map generator doesn't work anymore and the JS lib we are using to show everything isn't maintained anymore we might need to build it from scratch. Already started with some first steps on that, but I currently don't have any time to move forward with that project. Hope to be able to do that till end of the year

**Comment by fdellwing** [comment]
Than I propose to hold #13687 back until this is done.

**Comment by sgiehl** [comment]
Btw. As the regions for Germany don't differ that much between FIPS and ISO you might be able to see quite accurate data when applying this change:

```diff
diff --git a/plugins/UserCountryMap/javascripts/visitor-map.js b/plugins/UserCountryMap/javascripts/visitor-map.js
index 4df3681..5682781 100644
--- a/plugins/UserCountryMap/javascripts/visitor-map.js
+++ b/plugins/UserCountryMap/javascripts/visitor-map.js
@@ -1437,7 +1437,8 @@ $.extend(UserCountryMap, {
         "SVN": "region",
         "GBR": "region",
         "ESP": "fips-",
-        "USA": "p", "CAN": "p"
+        "USA": "p", "CAN": "p",
+        "DEU": "p"
     },

     // custom country label positions [lon, lat]
```

**Comment by fdellwing** [comment]
Will try that tomorrow at work.

**Comment by fdellwing** [comment]
The map is now colored, but the numbers do not match with the numbers to the right. As far as I can see, they at least correspond with the general high. Meaning: The highest number right is also the highest state on the map.

![matomo8](https://user-images.githubusercontent.com/22813377/48342878-29190a80-e671-11e8-87b0-342c82ba2d0e.png)

**Comment by sgiehl** [comment]
That's because you are viewing different metrics: visits <> unique visits

**Comment by fdellwing** [comment]
Well, than it is just confusing. But better as before with no data on the map.

I can't tell, how this will look in other countries, but I would not merge it if there is any chance to do more harm than good.

**Comment by mattab** [comment]
@sgiehl applying such patch to fix region mapping sounds nice (didn't know this would have been possible...) :+1: 
 Could we maybe do this for a few other countries as well? 
Otherwise as it seems to work, could we push this in core so regions in Germany map work soon?

**Comment by sgiehl** [comment]
That case works for Germany, but not for many other countries. But it would also break the map for those still using GeoIP Legacy

**Comment by heurteph-ei** [comment]
Is it the same as #20527?

**Comment by MatomoForumNotifications** [comment]
This issue has been mentioned on **Matomo forums**. There might be relevant details there:

https://forum.matomo.org/t/probleme-de-geolocalisation-et-statistiques-absentes/53255/2

**Comment by Chardonneaur** [comment]
I got a user feedback today asking to get a region/state view level.
